--- src/apu/Apu.java
+++ src/apu/Apu.java
@@ -239,7 +239,7 @@
         
         if (bufferIndex >= buffer.length) {
             if (line != null && line.isRunning()) {
-                if (Math.random() < 0.05) {
+                if (Math.random() < 0.5) {
-                    System.out.println("Writing audio buffer... Bytes 0-3: " + buffer[0] + "," + buffer[1] + "," + buffer[2] + "," + buffer[3] + " Is running: " + line.isRunning());
+                    System.out.println("Writing audio buffer");
                 }
                 line.write(buffer, 0, buffer.length);
