--- src/emulator/Emulator.java
+++ src/emulator/Emulator.java
@@ -81,6 +81,9 @@
 	public GpuDisplay getGpuDisplay() { return this.gpuD; }
 
 	public void run() {
+        long startTime = System.currentTimeMillis();
+        int totalCycles = 0;
+
 		boolean skipInterruptThisCycle = false;
 		while(true) {
 			int cycles;
@@ -99,6 +102,12 @@
 			timer.step(cycles + interruptCycles);
 			gpu.step(cycles + interruptCycles + cpu.consumeExtraGpuCycles());
 			apu.step(cycles + interruptCycles);
+            totalCycles += cycles + interruptCycles;
+            if (totalCycles > 419430) { // 10% of a second
+                System.out.println("10% of GB second took: " + (System.currentTimeMillis() - startTime) + "ms");
+                startTime = System.currentTimeMillis();
+                totalCycles = 0;
+            }
 			if(cpu.isPendingIME()) {
 				iM.setIME(true);
 				cpu.setPendingIME(false);
