--- src/cpu/Cpu.java
+++ src/cpu/Cpu.java
@@ -357,7 +357,7 @@
 		}
 		Instruction instruction = ins.get(op);
 		trace[traceIndex] = "PC: " + String.format("%04X", pc - 1) + ", Opcode: " + String.format("%02X", op);
-		if (cont % 1000000 == 0) System.out.println("PC: " + String.format("%04X", pc - 1));
+		if (cont % 10000 == 0) System.out.println("PC: " + String.format("%04X", pc - 1));
 		traceIndex = (traceIndex + 1) % trace.length;
 		if((pc>=0x8000 && pc<0xA000) || (pc>0xDFFF && pc<0xFF80)) {
 			oobCount++;
